sympy/functions/special/delta_functions.py

```diff
@@ -218,12 +234,25 @@ def eval(cls, arg):
             return S.Zero
         elif arg.is_positive:
             return S.One
+        elif arg.is_zero and H0 != S.NaN:
+            return H0
 
-    def _eval_rewrite_as_Piecewise(self, arg):
+    def _eval_rewrite_as_Piecewise(self, *args):
+        if len(args) == 2:
+            arg, H0 = args[0], args[1]
+        else:
+            arg, H0 = args[0], None
         if arg.is_real:
-            return Piecewise((1, arg > 0), (S(1)/2, Eq(arg, 0)), (0, True))
+            if H0 == None:
+                return Piecewise((1, arg > 0), (0, arg < 0))
```

Comment:
This would help users *discover* how to handle `Heaviside(0)`, which something I really like.  But I'm not sure how to implement it in a way consistent with Sympy's standard practices.

On a related note, it would also be good if we had a way of rewriting Heaviside's to introduce a default value into expressions where it was initially mixing, but I don't know how to do this either.